src/Action/DetermineNextRelease.php

```diff
@@ -49,24 +49,16 @@ public function __construct(
         $this->pullRequests = $pullRequests;
     }
 
-    public function __invoke(Project $project): NextRelease
+    public function __invoke(Project $project, Branch $branch): NextRelease
     {
         $repository = $project->repository();
 
         try {
-            $branch = $project->stableBranch() ?? $project->unstableBranch();
-        } catch (NoBranchesAvailable $e) {
-            throw CannotDetermineNextRelease::forProject(
-                $project,
-                $e
-            );
-        }
-
-        try {
-            $currentRelease = $this->releases->latest($repository);
+            $currentRelease = $this->releases->branchLatest($repository, $branch);
```

Comment:
```suggestion
            $currentRelease = $this->releases->latestBranch($repository, $branch);
```
(I think)

Reply:
The method fetch latest release from given branch, maybe should simply be latestRelease, or more verbose latestBranchRelease

Reply:
There is a `latest` method. It would be weird to add `Release` to this method and not the `latest` one.

Here I wanted to say it's the branch's latest release. So branchLatest.
Maybe latestForBranch is more clear ?